src/Sema.zig

```diff
@@ -7720,6 +7724,41 @@ fn funcCommon(
             });
         };
 
+        // If the return type is comptime only but not dependent on parameters then all parameter types also need to be comptime
+        if (!is_dependent_ret_ty and try sema.typeRequiresComptime(block, ret_ty_src, return_type)) {
```

Comment:
`is_dependent_ret_ty` should always be the same as `ret_poison`.

Reply:
They do not always seem to be the same, and the test cases do not build when using `ret_poison` instead of `is_dependent_ret_ty`.

Reply:
Hmm, using `ret_poison` does seem to break something but the current solution is unable to handle even simple cases like:
```zig
const A = comptime_int;
fn foo(a: u32) A {
    _ = a;
    return undefined;
}
comptime {
    _ = foo;
}
```
Maybe it'd be better to just scrap this error altogether along with the broken `parameter of type 'T' must be declared comptime`...